### PR TITLE
Endpoint routes renamed + various additionals

### PR DESCRIPTION
**Pull request by Ridderrasmus**

### CODE REVIEW

**Comment by Jhm20P** [comment]
All endpoints to be async

File: API/Controllers/roomController.cs [hunk 1]
```diff
@@ -150,5 +152,42 @@ public IActionResult Search([FromQuery] SearchRoomQuery query )
...
             
         }
+
+        /// <summary>
+        /// Get the details of a room as a nice to digest DTO
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpGet("RoomDetails/{id}")]
+        public IActionResult GetRoomDetails([FromRoute] int id)
```
Comment (Jhm20P) [inline API/Controllers/roomController.cs:162]: All endpoints have to be async

File: API/Controllers/ReservationsController.cs [hunk 2]
```diff
@@ -34,7 +34,7 @@ public ReservationsController(HotelContext context, UserManager<User> userManage
         /// </summary>
         /// <returns>Status OK with the list of reservations</returns>
         [HttpGet]
-        public IActionResult Get()
+        public async Task<IActionResult> Get()
         {
             // Currently the simplest CRUD operation
             var reservations = _context.Reservations.ToList();
```
Comment (Jhm20P) [inline API/Controllers/ReservationsController.cs:40]: Shouldn't it be var reservations = await _context.Reservations.ToListAsync();

File: API/Controllers/TicketController.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ public TicketController(HotelContext context)
         }
         //fetch all tickets
         [HttpGet]
-        public IActionResult GetAll()
+        public async Task<IActionResult> GetAll()
         {
             var ticket = _context.Tickets.ToList();
```
Comment (Jhm20P) [inline API/Controllers/TicketController.cs:30]: Shouldn't it be ticket = await _context.Tickets.ToListAsync();

File: API/Controllers/ReservationsController.cs [hunk 4]
```diff
@@ -47,7 +47,7 @@ public IActionResult Get()
         /// <param name="id">Reservation ID</param>
         /// <returns>Status OK with reservation</returns>
         [HttpGet("{id}")]
-        public IActionResult Get(int id)
+        public async Task<IActionResult> Get(int id)
         {
             // Currently the simplest CRUD operation
             var reservation = _context.Reservations.Find(id);
```
[the remark this answers is not in the document]
Comment (Ridderrasmus) [reply API/Controllers/ReservationsController.cs:53]: Good point. Should ideally be changed as well. Doing a new pass.

File: API/Controllers/TicketController.cs [hunk 5]
```diff
@@ -38,7 +38,7 @@ public IActionResult GetAll()
         }
         //fetch the id of a ticket
         [HttpGet("{id}")]
-        public IActionResult GetById([FromRoute] int id)
+        public async Task<IActionResult> GetById([FromRoute] int id)
         {
             var ticket = _context.Tickets.Find(id);
```
Comment (Jhm20P) [inline API/Controllers/TicketController.cs:43]: Shouldn't it be var ticket = await _context.Tickets.ToFindAsync(id);

File: API/Controllers/roomController.cs [hunk 6]
```diff
@@ -27,7 +27,7 @@ public RoomController(HotelContext context)
         /// </summary>
         /// <returns>Status OK with the list of rooms</returns>
         [HttpGet]
-        public IActionResult GetAll() 
+        public async Task<IActionResult> GetAll() 
         {
             // Currently the simplest CRUD operation
             var room = _context.Rooms.ToList();
```
Comment (Jhm20P) [inline API/Controllers/roomController.cs:33]: Shouldn't it be var room = await _context.Rooms.ToListAsync();

File: API/Controllers/roomController.cs [hunk 7]
```diff
@@ -91,7 +91,7 @@ public IActionResult Update([FromBody] Room room)
         /// <returns>Status OK</returns>
         [HttpDelete("{id}")]
         [Authorize(Roles = "Admin")]
-        public IActionResult Delete(int id)
+        public async Task<IActionResult> Delete(int id)
         {
             // Currently the simplest CRUD operation
             var room = _context.Rooms.Find(id);
```
Comment (Jhm20P) [inline API/Controllers/roomController.cs:97]: Shouldn't it be var room = await _context.RoomsToFindAsync(id);

File: API/Controllers/roomController.cs [hunk 8]
```diff
@@ -159,7 +159,7 @@ public IActionResult Search([FromQuery] SearchRoomQuery query )
         /// <param name="id"></param>
         /// <returns></returns>
         [HttpGet("RoomDetails/{id}")]
-        public IActionResult GetRoomDetails([FromRoute] int id)
+        public async Task<IActionResult> GetRoomDetails([FromRoute] int id)
         {
             var room = _context.Rooms.Find(id);
```
Comment (Jhm20P) [inline API/Controllers/roomController.cs:164]: Shouldn't it be var room = await _context.Rooms.ToFindAsync(id);

File: API/Controllers/roomController.cs [hunk 9]
```diff
@@ -184,7 +184,7 @@ public IActionResult GetRoomDetails([FromRoute] int id)
         /// </summary>
         /// <returns></returns>
         [HttpGet("RoomTypes")]
-        public IActionResult GetRoomTypes()
+        public async Task<IActionResult> GetRoomTypes()
         {
             var roomTypes = _context.RoomTypes.ToList();
```
Comment (Jhm20P) [inline API/Controllers/roomController.cs:189]: Shouldn't it be var roomTypes = await _context.RoomTypes.ToListAsync();

**Review by Jhm20P — APPROVED** [review]

**Pull request merged**